client.go

```diff
@@ -1650,6 +1652,10 @@ func (f *File) ReadFrom(r io.Reader) (int64, error) {
 			}
 		}
 
+		if remain < 0 {
+			remain = math.MaxInt64
+		}
```

Comment:
This was added from discussion in #426 where we want to default to maximum concurrency when we have a `Size` reported as less than 0.